tests/test_helpers.py

```diff
@@ -66,3 +70,280 @@ def test_local_model_non_default_name(self, tmp_path):
+
+        with torch.no_grad():
+            logits_after_scaling = model(**inputs).logits
+
+        assert torch.allclose(logits_before_scaling, logits_after_scaling)
+
+    def test_wrong_scaling_datatype(self):
+        model = AutoModelForCausalLM.from_pretrained("facebook/opt-125m")
+        lora_config = LoraConfig(
+            r=4,
+            lora_alpha=4,
+            target_modules=["k_proj", "v_proj"],
+            lora_dropout=0.1,
+            bias="none",
+            init_lora_weights=False,
+        )
+
+        model = get_peft_model(model, lora_config)
+        model.eval()
+        tokenizer = AutoTokenizer.from_pretrained("facebook/opt-125m")
+        inputs = tokenizer("hello world", return_tensors="pt")
+
+        with torch.no_grad():
+            logits_before_scaling = model(**inputs).logits
+
+        scales_before_scaling = self.get_scale_from_modules(model)
+
+        # we expect a type error here becuase of wrong datatpye of alpha
+        alpha = "a"
+        with pytest.raises(TypeError, match=f"{alpha} should be of type float, got {type(alpha)}"):
+            with set_adapter_scale(model=model, alpha=alpha):
+                pass
+
+        scales_after_scaling = self.get_scale_from_modules(model)
+        for key in scales_before_scaling.keys():
+            assert scales_before_scaling[key] == scales_after_scaling[key]
+
+        with torch.no_grad():
+            logits_after_scaling = model(**inputs).logits
+
+        assert torch.allclose(logits_before_scaling, logits_after_scaling)
+
+    def test_not_lora_model(self):
+        model = AutoModelForCausalLM.from_pretrained("facebook/opt-125m")
+
+        # we expect a value error here because the model
+        # does not have lora layers
+        with pytest.raises(ValueError, match="scaling is only supported for models with `LoraLayer`s"):
+            with set_adapter_scale(model=model, alpha=0.5):
+                pass
+
+    def test_no_scaling(self):
+        base_model = AutoModelForCausalLM.from_pretrained("facebook/opt-125m")
+        tokenizer = AutoTokenizer.from_pretrained("facebook/opt-125m")
+        inputs = tokenizer("hello world", return_tensors="pt")
+
+        base_model.eval()
+
+        with torch.no_grad():
+            logits_base_model = base_model(
+                **inputs,
+            ).logits
+
+        lora_config = LoraConfig(
+            r=4,
+            lora_alpha=4,
+            target_modules=["k_proj", "v_proj"],
+            lora_dropout=0.1,
+            bias="none",
+            init_lora_weights=False,
+        )
+        lora_model = get_peft_model(base_model, lora_config)
+        lora_model.eval()
+
+        with set_adapter_scale(model=lora_model, alpha=0.0):
+            with torch.no_grad():
+                logits_lora_model = lora_model(
+                    **inputs,
+                ).logits
+
+        assert torch.allclose(logits_base_model, logits_lora_model)
+
+    def test_diffusers_pipeline(self):
```

Comment:
Nice! Would you mind adding a similar one for transformers models loaded directly with PEFT? See: https://huggingface.co/docs/transformers/v4.43.3/en/peft#load-adapters-with--peft

Reply:
Correct me if I am wrong, but with a directly loaded model from PEFT, there should not be any LoraLayer to check for. And that should mean, we will not be able to access the `scaling` attribute.

How would you want me to approach this?

Reply:
This should work. To give an example:

```python
# first create a peft checkpoint
model_id = "facebook/opt-125m"
model = AutoModelForCausalLM.from_pretrained(model_id)
config = LoraConfig(init_lora_weights=False)
model = get_peft_model(model, config)
model.save_pretrained("/tmp/peft/opt-lora")
del model

# load directly into transformers model
model = AutoModelForCausalLM.from_pretrained(model_id)
model.load_adapter("/tmp/peft/opt-lora")
type(model.model.decoder.layers[0].self_attn.v_proj)  # => peft.tuners.lora.layer.Linear
```